src/gui/gui-nick.c

```diff
@@ -57,7 +59,21 @@ gui_nick_hash_color (const char *nickname)
     if (config_num_nick_colors == 0)
         return 0;
 
-    ptr_nick = nickname;
+    length = strlen (CONFIG_STRING(config_look_nick_color_hash_seed)) +
```

Comment:
Add conditional to only do the `malloc` and salt concatenation when the salt is non-empty. Avoids unnecessary allocations and string operations by default and in most cases.